crates/starknet-devnet-core/src/state/mod.rs

```diff
@@ -50,12 +50,13 @@ pub trait CustomState {
         contract_class: ContractClass,
     ) -> DevnetResult<()>;
 
-    /// Link contract address to class hash
-    fn predeploy_contract(
+    /// Link contract address to class hash. Does not include balance modification or constructor
+    /// simulation. Returns `true` if deployment was performed by this method, `false` otherwise.
+    fn predeploy_if_undeployed(
```

Comment:
Consider reverting this rename

Reply:
https://github.com/0xSpaceShard/starknet-devnet/pull/799/commits/dee7d572cd0297e6a516f747168c478c1026348b